python/pyarrow/_parquet.pyx

```diff
@@ -1083,6 +1084,50 @@ cdef class ParquetReader:
     def set_use_threads(self, bint use_threads):
         self.reader.get().set_use_threads(use_threads)
 
+    def iter_batches(self, int64_t batch_size, row_groups, column_indices=None,
```

Comment:
Ditto, I'd add a method `def set_batch_size(self, int64_t batch_size)`. Why do you need the `batch_size` and `use_threads` if they're already property of the class and the caller can change them via setters of the instance?

Reply:
I have those parameters because I was following the pattern set by the [`read_row_group()` and `read_row_groups()`](https://github.com/apache/arrow/blob/66a18e14f64359b73bd8ed5625c3092cd369af44/python/pyarrow/_parquet.pyx#L1131-L1136) methods in the same file, which both have a `use_threads` parameters. Same with the `read_all()` method.

Reply:
Noted for consistency. I find it un-intuitive that calling this method changes the batch size (or threading) for the following callers. Since this is not introduced by this PR, I won't make this a blocking issue.

I'll leave @jorisvandenbossche decide if we should have a followup to fix this anti-pattern.